FHEM/14_SD_WS07.pm

```diff
@@ -181,45 +166,100 @@ SD_WS07_Parse($$)
 	
 	#Log3 $name, 4, "$iohash->{NAME} SD_WS07: $name ($rawData)";  
 
-	if (!defined(AttrVal($hash->{NAME},"event-min-interval",undef)))
-	{
+	if (!defined(AttrVal($hash->{NAME},"event-min-interval",undef))) {
 		my $minsecs = AttrVal($iohash->{NAME},'minsecs',0);
 		if($hash->{lastReceive} && (time() - $hash->{lastReceive} < $minsecs)) {
-			Log3 $hash, 4, "$iohash->{NAME} SD_WS07: $name $deviceCode dropped due to short time. minsecs=$minsecs";
+			Log3 $hash, 4, "$deviceCode Dropped due to short time. minsecs=$minsecs";
 		  	return "";
 		}
 	}
 	
 	$hum += AttrVal($name, "correction-hum", 0);				# correction value for humidity (default 0 %)
 	if ($hum > 99) {
-		Log3 $name, 4, "$iohash->{NAME} SD_WS07: $name ERROR - Humidity unknown ($hum)";
+		Log3 $name, 3, "$iohash->{NAME}: $name ERROR - Humidity unknown ($hum)";
 		return "";
 	}
 	
+   if ($temp > 700 && $temp < 3840) {								# -25,6 .. 70,0 °C
+		Log3 $name, 3, "$iohash->{NAME}: $name: ERROR - Temperature unknown ($temp)";
+		return "";
+   } elsif ($temp >= 3840) {        # negative Temperaturen, ist ueberprueft worden
+      $temp -= 4096;
+   }  
+   $temp /= 10;
 	$temp += AttrVal($name, "correction-temp", 0);				# correction value for temperature (default 0 K)
-	Log3 $name, 4, "$iohash->{NAME} SD_WS07: $name id=$id, channel=$channel, temp=$temp, hum=$hum, bat=$bat";
+   Log3 $iohash, 4, "$iohash->{NAME}: $name id=$id, channel=$channel, temp=$temp, hum=$hum, bat=$bat";
 
+	# Sanity check temperature and humidity
+   if($def) {
+		my $timeSinceLastUpdate = ReadingsAge($hash->{NAME}, "state", 0);
+		if ($timeSinceLastUpdate < 0) {
+			$timeSinceLastUpdate *= -1;
```

Comment:
$timeSinceLastUpdate *= -$timeSinceLastUpdate *; ist vielleicht einfacher lesbar oder geht das so in perl nicht?

Reply:
Deine Formel akzeptiert Perl nicht, oder ist "*" am Ende ein Tippfehler? Ich würde dann vielleicht besser gleich folgendes verwenden:

		my $timeSinceLastUpdate = abs(ReadingsAge($hash->{NAME}, "state", 0));

Reply:
* ist ein tippfehler. 
$timeSinceLastUpdate = -$timeSinceLastUpdate;

Reply:
Wieder was gelernt, die Schreibweise kannte ich noch nicht.